src/Core/Services/AirtelMoneyService.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Text.RegularExpressions;
+using Core.Entities;
+
+namespace Core.Services
+{
+    public class AirtelMoneyService : IPaymentService
+    {
+        public readonly string _message;
+        public readonly string _phoneNumber;
+
+        public AirtelMoneyService(string message, string phoneNumber)
+        {
+            _message = IPaymentService.SanitizeMessage(message);
+            _phoneNumber = phoneNumber;
+        }
+
+        public Payment GeneratePayment()
+        {
+            var amountRegex = new Regex("((?<=(Amount: )|(Amt: ))(.*?)(?=M))|((?<=(recieved ))(.*?)(?=M))");
+            var referenceRegex = new Regex(@"(?<=Ref: )(.*?)(?=\s)");
+            var bankNameRegex = new Regex("(?<=from )(.*?)(?= on)");
+            var fromAgentRegex = new Regex("(Cash In)");
+            var amount = Decimal.Parse(amountRegex.Match(_message).ToString().Trim());
+            var reference = referenceRegex.Match(_message).ToString();
+            var fromAgent = fromAgentRegex.IsMatch(_message);
+            var BankName = bankNameRegex.Match(_message).ToString();
+
+            return new Payment()
+            {
+                Amount = amount,
+                PhoneNumber = _phoneNumber,
+                FromAgent = fromAgent,
```

Comment:
this should be `string AgentName` not FromAgent and indicate whether its `None` or `Missing` depending on the circumstance